app/store/sagas/clubs.js

```diff
@@ -29,22 +28,20 @@ function* listMembers() {
   // TODO: this should not be done this way! (current club ID)
   const clubID = 288750;
   yield put(setCurrentClubID("loading"));
-  const { ids, error } = yield call(listClubMembers, accessToken, clubID);
+  const { ids, error, entities } = yield call(
+    listClubMembers,
+    accessToken,
+    clubID
+  );
   if (!error) {
     yield put(updateEntity(clubID, "clubs", { members: ids }));
-    const filteredIds = ids.filter(id => id !== currentUserID);
-    let mergedEntities = {};
-    for (let index = 0; index < filteredIds.length; index += 1) {
-      const { entities } = yield getStats(filteredIds[index]);
-      mergedEntities = {
-        ...mergedEntities,
-        athletes: {
-          ...mergedEntities.athletes,
-          ...entities.athletes
-        }
-      };
-    }
-    yield put(setEntity("athletes", mergedEntities.athletes));
+
+    const athletes = Object.keys(entities.members)
```

Comment:
you should extract l39- l42 inside in a function
to make your saga more readable and testable

Reply:
service: 

export const filterClubAthletes = (athletes, filteredAthleteID) => {
  return Object.keys(athletes)
    .filter(member => parseInt(member, 0) !== filteredAthleteID)
    .map(key => ({ [key]: athletes[key] }))
    .reduce((accumulator, current) => ({ ...accumulator, ...current }), {});
};

saga: 

 const athletes = yield call(
      filterClubAthletes,
      entities.members,
      currentUserID
    );